### ISSUE TITLE
Retrieve OAuth credentials for additional scopes from HangoutApi object

### ISSUE THREAD

**Issue by polux** [Scarygami/dart_hangouts_api#1]
In the devconsole config page for hangouts you can specify additional OAuth scopes to be requested alongside the Hangout API scopes, but there's then no straightforward way to retrieve them and reuse them with [googleapis_auth](https://pub.dartlang.org/packages/googleapis_auth). It would be a nice feature to have.

**Comment by Scarygami** [comment]
The way this works with the JS-client is to use

    gapi.auth.authorize({
      client_id: null,
      scope: scopes,
      immediate: isImmediate
    }, callback);

Similar to [`createImplicitBrowserFlow`](https://github.com/dart-lang/googleapis_auth/blob/master/lib/auth_browser.dart#L54) it would be easy to create a `BrowserOAuth2Flow` using an `ImplicitFlow` with the `clientId` set to `null`, and the scopes matching the ones you requested plus the default hangout scopes that are requested automatically: `https://www.googleapis.com/auth/plus.me`,
`https://www.googleapis.com/auth/hangout.av`,
`https://www.googleapis.com/auth/hangout.participants`

I'll add an according helper-function plus explanation to this library. Thanks for the suggestion :)

**Comment by polux** [comment]
Cool, thanks!

**Comment by Scarygami** [comment]
I tried to implement this and first tests seem promising.
Currently not on pub but if you could change your dependency to point to this repo and give it a try it would be nice :)

```
import "package:hangouts_api/hangouts_auth.dart";
import 'package:googleapis/plus/v1.dart'; // or any API you want to use

function login() {
    // add any additional scopes here, the default hangout scopes will be added automatically
    var scopes = []; 

    createHangoutAuthFlow(scopes).then((flow) {
        flow.clientViaUserConsent().then((client) {
          flow.close();
          var plus = new PlusApi(client);
          plus.people.get('me').then(...);
        });
    });
}
```

**Comment by Scarygami** [comment]
I've published this as 0.3.0 now since it works for me.
Also added a paragraph about it to the readme.

Let me know/open a new issue if you run into any problems, thanks :)

**Comment by polux** [comment]
Sorry, haven't had a chance to test it yet. I'll let you know. Thanks for the quick implementation!
